presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/DynamicMetastoreConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.thrift;
+
+import com.facebook.airlift.configuration.Config;
+import com.facebook.airlift.configuration.ConfigDescription;
+
+public class DynamicMetastoreConfig
+{
+    private String metastoreUsername;
```

Comment:
Can we merge the two XXXMetastoreConfig into one? `metastoreUsername` will have conflict

Reply:
2 questions
1. Can you help me understand why would there be a conflict? Presto server can be bootstrapped either with Static or Dynamic HMS cluster, and either Static or Dynamic config will be instantiated respectively. So how would a conflict arise?
2. If we merge both the MetastoreConfig classes, does that mean either hive.metastore.uri or hive.metastore.discovery.uri, one of them would be null based on the type of cluster specified?